solr/core/src/java/org/apache/solr/handler/admin/ClusterStatus.java

```diff
@@ -92,11 +96,57 @@ public static Health combine(Collection<Health> states) {
   public ClusterStatus(ZkStateReader zkStateReader, ZkNodeProps props) {
     this.zkStateReader = zkStateReader;
     this.message = props;
-    collection = props.getStr(ZkStateReader.COLLECTION_PROP);
+    collectionParam = props.getStr(ZkStateReader.COLLECTION_PROP);
+    liveNodesParam = props.getBool(ZkStateReader.LIVENODES_PROP, false);
+    clusterPropertiesParam = props.getBool(ZkStateReader.CLUSTER_PROP, false);
+    rolesParam = props.getBool(ZkStateReader.ROLES_PROP, false);
+    includeAll = props.getBool(ZkStateReader.INCLUDE_ALL, true);
```

Comment:
Add a new property to CLUSTERSTATUS request to fetch all info - live nodes, collections, aliases, roles, cluster properties. Set to true by default.